packages/utils/src/mergeDefaultsWithFormData.ts

```diff
@@ -41,7 +41,7 @@ export default function mergeDefaultsWithFormData<T = any>(
     const overrideOppositeArray = overrideFormDataWithDefaults ? formData : defaultsArray;
 
     const mapped = overrideArray.map((value, idx) => {
-      if (overrideOppositeArray[idx]) {
+      if (typeof overrideOppositeArray[idx] !== 'undefined') {
```

Comment:
```suggestion
     // We want to explicitly make sure that the value is NOT undefined since null, 0 and empty space are valid values
      if (overrideOppositeArray[idx] !== undefined) {
```